### PR TITLE
Add a ButtonModule class

### PR DESCRIPTION
**Pull request by KaiVolland**
The buttonmodule contains information about apperance and action.

The action of the button is define via enum. It address the buttons contained in
the BasiGX Package (https://github.com/terrestris/BasiGX/tree/master/src/view/button).

Additional there are two more generic types:
* "OPENMODULEWINDOW" (property connectedModule required)
    * Opens any module in an Ext.window.Window.
* "TOGGLEINTERACTION" (property interaction required)
    * Toggles the functionality of an OpenLayers 3 interaction.

### CODE REVIEW

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Button.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+		ADDWMS("addwms"),
+		COORDINATETRANSFORM("coordinatetransform"),
+		HELP("help"),
+		HSI("hsi"),
+		MEASURELINE("measureline"),
+		MEASUREARE("measurearea"),
+		PERMALINK("permalink"),
+		TOGGLELEGEND("togglelegend"),
+		ZOOMIN("zoomin"),
+		ZOOMOUT("zoomut"),
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Button.java]: I think this should be ZOOMOUT("zoomout")

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Button.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+	 */
+	public static enum ButtonActionType {
+		OPENMODULEWINDOW("openmodulewindow"),
+		TOGGLEINTERACTION("toggleinteraction"),
+		ADDWMS("addwms"),
+		COORDINATETRANSFORM("coordinatetransform"),
+		HELP("help"),
+		HSI("hsi"),
+		MEASURELINE("measureline"),
+		MEASUREARE("measurearea"),
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/module/Button.java]: I think this should be MEASUREAREA("measurearea")

**Comment by buehner** [comment]
Thx @KaiVolland 

The button seems nice,but i think the enum ButtonActionType should be refactored to a String ButtonActionType as we currently can not be sure that this will be a fix set of values.

We should also fix the coveralls problem before merging this.

**Comment by KaiVolland** [comment]
Refactored the ButtonActionType.

**Comment by marcjansen** [comment]
Please change the PR title to e.g. "Add a ButtonModule class"

**Pull request merged**